src/VirtoCommerce.ElasticAppSearch.Data/Services/Converters/DocumentConverter.cs

```diff
@@ -49,11 +52,16 @@ public virtual (Document, Schema) ToProviderDocument(IndexDocument indexDocument
                 document.Fields.Add(fieldName, field.IsCollection ? field.Values : field.Value);
                 schema.Fields.Add(fieldName, ToProviderFieldType(field));
             }
+
+            if (field.Name == ModuleConstants.Api.FieldNames.ObjectFieldName && document.Fields.ContainsKey(fieldName))

```

Comment:
Ok as temporary workaround, but it's better to find the way to prevent use snake case as serialization rule for all document fields